mondrian/src/main/java/mondrian/olap/fun/CrossJoinFunDef.java

```diff
@@ -999,29 +999,27 @@ private static boolean checkData(
         Evaluator evaluator)
     {
         if (cnt < 0) {
-            // no measures found, use standard algorithm
-            if (measureSet.isEmpty()) {
-                Object value = evaluator.evaluateCurrent();
-                if (value != null
-                    && !(value instanceof Throwable))
-                {
-                    return true;
-                }
-            } else {
+            boolean found = false;
+            Object value = evaluator.evaluateCurrent();
+            if (value != null
+                && !(value instanceof Throwable))
+            {
+                found = true;
+            }
+            if (!measureSet.isEmpty()) {
```

Comment:
Took a closer look this morning.  I'm worried about the performance implications of this.  The algorithm formerly reduced the evaluation required by explicitly looking only at base measures.  By forcing full evaluation I suspect that some users with higher cardinality dimensions will be badly impacted.  The purpose of the crossjoin optimizer is to reduce the evaluation space by determining which tuples are likely to have non-empty data given the presence of base measure data.  Evaluating calculated members effectively removes the value of the optimizer.

I'd like to think about this problem and see if I can come up with a reasonable alternative.